packages/react-native/src/immutable-form/actions.js

```diff
@@ -38,8 +38,14 @@ export const submitForm = (formName: string) => (submit: (Object) => Promise<any
     const state = getState();
 
     submit(state).then(
-      res => dispatch(approvedForm(formName)(res)),
-      err => dispatch(errorForm(formName)(errorMessage(err))),
+      res => {
+        dispatch(clearForm(formName)());
+        dispatch(approvedForm(formName)(res));
+      },
+      err => {
+        dispatch(clearForm(formName)());
```

Comment:
We actually don't want to clear the form when there is an error as the form has a field called `error` which will not be shown if this is done this way. Also, if the user typed a wrong email / username, we want to let him try to correct the typo